sdk/java/pulumi/src/test/java/com/pulumi/resources/CustomTimeoutsTest.java

```diff
@@ -31,4 +32,67 @@ Stream<DynamicNode> testGolangString() {
         );
     }
 
+    @TestFactory
+    Stream<DynamicNode> testParseTimeoutString() {
+        return Stream.of(
+            // Basic cases
+            dynamicTest("null input", () ->
+                    assertThat(CustomTimeouts.parseTimeoutString(null)).isNull()),
+            dynamicTest("empty input", () ->
+                    assertThat(CustomTimeouts.parseTimeoutString("")).isNull()),
+            dynamicTest("zero", () ->
+                    assertThat(CustomTimeouts.parseTimeoutString("0")).isEqualTo(Duration.ZERO)),
```

Comment:
```suggestion
                    assertThat(CustomTimeouts.parseTimeoutString("0")).isEqualTo(Duration.ZERO)),
                    assertThat(CustomTimeouts.parseTimeoutString("0s")).isEqualTo(Duration.ZERO)),
```

I think this should just work, but might be worth an additional test case.